frontend/src/components/Pages/InspectionPage/InspectionSection.tsx

```diff
@@ -60,26 +117,84 @@ export interface DeckMissionType {
     }
 }
 
-export interface OngoingMissionType {
+export interface DeckMissionCount {
+    [color: string]: {
+        count: number
+        message: string
+    }
+}
+
+export interface ScheduledMissionType {
     [missionId: string]: boolean
 }
 
+interface IInspectionProps {
+    refreshInterval: number
+    updateScheduledMissionsMap: (areaMissions: DeckMissionType) => Promise<void>
+    scheduledMissions: ScheduledMissionType
+    ongoingMissions: ScheduledMissionType
+}
+
+const getDeadlineInspection = (deadline: Date) => {
+    const deadlineDays = getDeadlineInDays(deadline)
+    switch (true) {
+        case deadlineDays <= 1:
+            return 'red'
+        case deadlineDays > 1 && deadlineDays <= 7:
+            return 'red'
+        case deadlineDays > 7 && deadlineDays <= 14:
+            return 'orange'
+        case deadlineDays > 7 && deadlineDays <= 30:
+            return 'green'
+    }
+    return 'green'
+}
+
 export const compareInspections = (i1: Inspection, i2: Inspection) => {
     if (!i1.missionDefinition.inspectionFrequency) return 1
     if (!i2.missionDefinition.inspectionFrequency) return -1
-    if (!i1.missionDefinition.lastRun) return 1
-    if (!i2.missionDefinition.lastRun) return -1
+    if (!i1.missionDefinition.lastRun) return -1
+    if (!i2.missionDefinition.lastRun) return 1
     else return i1.deadline!.getTime() - i2.deadline!.getTime()
 }
 
-export function InspectionSection({ refreshInterval }: RefreshProps) {
+export function InspectionSection({
+    refreshInterval,
+    updateScheduledMissionsMap,
+    scheduledMissions,
+    ongoingMissions,
+}: IInspectionProps) {
     const { TranslateText } = useLanguageContext()
     const { installationCode } = useInstallationContext()
     const [deckMissions, setDeckMissions] = useState<DeckMissionType>({})
-    const [ongoingMissions, setOngoingMissions] = useState<OngoingMissionType>({})
     const [selectedDeck, setSelectedDeck] = useState<Deck>()
     const [selectedMissions, setSelectedMissions] = useState<CondensedMissionDefinition[]>()
-    const [isDialogOpen, setisDialogOpen] = useState<boolean>(false)
+    const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false)
+    const [unscheduledMissions, setUnscheduledMissions] = useState<CondensedMissionDefinition[]>([])
+    const [isAlreadyScheduled, setIsAlreadyScheduled] = useState<boolean>(false)
+
+    const openDialog = () => {
+        setIsDialogOpen(true)
+    }
+
+    const closeDialog = () => {
+        setIsAlreadyScheduled(false)
+        setSelectedMissions([])
+        setUnscheduledMissions([])
+        setIsDialogOpen(false)
+    }
+
+    useEffect(() => {
+        if (selectedMissions) {
+            let unscheduledMissions: CondensedMissionDefinition[] = []
+            selectedMissions!.forEach((mission) => {
```

Comment:
```suggestion
        if (selectedMissions) {
            let unscheduledMissions: CondensedMissionDefinition[] = []
            selectedMissions.forEach((mission) => {
```
Not necessary with ! if you already have a check to see if its empty